File: android-app/app/src/main/java/com/example/minitasktracker/core/ui/components/AppSnackbar.kt
package com.example.minitasktracker.core.ui.components

import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.dp
import com.example.minitasktracker.core.ui.theme.*

/**
 * TekTech Snackbar Types
 */
enum class SnackbarType {
  SUCCESS,
  ERROR,
  WARNING,
  INFO
}

/**
 * TekTech AppSnackbar
 * 
 * Consistent notification/toast component
 * - Success (green)
 * - Error (red)
 * - Warning (amber)
 * - Info (cyan)
 * 
 * @param message Snackbar message
 * @param type Snackbar type (success/error/warning/info)
 * @param actionLabel Optional action button text
 * @param onActionClick Optional action callback
 * @param onDismiss Dismiss callback
 */
@Composable
fun AppSnackbar(
  message: String,
  type: SnackbarType = SnackbarType.INFO,
  actionLabel: String? = null,
  onActionClick: (() -> Unit)? = null,
  onDismiss: () -> Unit
) {
  val (icon, containerColor, contentColor) = when (type) {
    SnackbarType.SUCCESS -> Triple(
      Icons.Default.CheckCircle,
      Green600,
      White
    )
    SnackbarType.ERROR -> Triple(
      Icons.Default.Error,
      Red600,
      White
    )
    SnackbarType.WARNING -> Triple(
      Icons.Default.Warning,
      Amber600,
      White
    )
    SnackbarType.INFO -> Triple(
      Icons.Default.Info,
      Cyan600,
      White
    )
  }
  
  Snackbar(
    modifier = Modifier.padding(Spacing.spacing16),
    action = if (actionLabel != null && onActionClick != null) {
      {
        TextButton(onClick = onActionClick) {
          Text(
            text = actionLabel,
            color = contentColor
          )
        }
      }
    } else {
      {
        IconButton(onClick = onDismiss) {
          Icon(
            imageVector = Icons.Default.Close,
            contentDescription = "Dismiss",
            tint = contentColor
          )
        }
      }
    },
    containerColor = containerColor,
    contentColor = contentColor,
    shape = MaterialTheme.shapes.medium
  ) {
    Row(
      verticalAlignment = Alignment.CenterVertically,
      horizontalArrangement = Arrangement.spacedBy(Spacing.spacing12)
    ) {
      Icon(
        imageVector = icon,
        contentDescription = null,
        modifier = Modifier.size(Sizes.iconSm)
      )
      Text(
        text = message,
        style = MaterialTheme.typography.bodyMedium
      )
    }
  }
}

/**
 * Helper function to show success snackbar
 */
@Composable
fun SuccessSnackbar(
  message: String,
  onDismiss: () -> Unit
) {
  AppSnackbar(
    message = message,
    type = SnackbarType.SUCCESS,
    onDismiss = onDismiss
  )
}

/**
 * Helper function to show error snackbar
 */
@Composable
fun ErrorSnackbar(
  message: String,
  actionLabel: String? = "Tekrar Dene",
  onActionClick: (() -> Unit)? = null,
  onDismiss: () -> Unit
) {
  AppSnackbar(
    message = message,
    type = SnackbarType.ERROR,
    actionLabel = actionLabel,
    onActionClick = onActionClick,
    onDismiss = onDismiss
  )
}

File: flutter_app/lib/core/widgets/app_snackbar.dart
import 'package:flutter/material.dart';
import '../theme/app_theme.dart';

/// TekTech Snackbar Types
enum SnackbarType { success, error, warning, info }

/// TekTech AppSnackbar
///
/// Consistent notification/toast component
/// - Success (green)
/// - Error (red)
/// - Warning (amber)
/// - Info (cyan)
class AppSnackbar {
  /// Show a custom snackbar
  static void show({
    required BuildContext context,
    required String message,
    SnackbarType type = SnackbarType.info,
    String? actionLabel,
    VoidCallback? onActionPressed,
    Duration duration = const Duration(seconds: 4),
  }) {
    final (icon, containerColor, contentColor) = _getColors(type);

    ScaffoldMessenger.of(context).showSnackBar(
      SnackBar(
        content: Row(
          children: [
            Icon(icon, color: contentColor, size: 20),
            const SizedBox(width: 12),
            Expanded(
              child: Text(
                message,
                style: TextStyle(
                  color: contentColor,
                  fontSize: 14,
                  fontWeight: FontWeight.w500,
                ),
              ),
            ),
          ],
        ),
        backgroundColor: containerColor,
        behavior: SnackBarBehavior.floating,
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(AppTheme.radius12),
        ),
        action: actionLabel != null && onActionPressed != null
            ? SnackBarAction(
                label: actionLabel,
                textColor: contentColor,
                onPressed: onActionPressed,
              )
            : null,
        duration: duration,
        margin: const EdgeInsets.all(16),
      ),
    );
  }

  /// Show success snackbar
  static void showSuccess({
    required BuildContext context,
    required String message,
    Duration duration = const Duration(seconds: 3),
  }) {
    show(
      context: context,
      message: message,
      type: SnackbarType.success,
      duration: duration,
    );
  }

  /// Show error snackbar
  static void showError({
    required BuildContext context,
    required String message,
    String? actionLabel,
    VoidCallback? onActionPressed,
    Duration duration = const Duration(seconds: 5),
  }) {
    show(
      context: context,
      message: message,
      type: SnackbarType.error,
      actionLabel: actionLabel ?? 'Tekrar Dene',
      onActionPressed: onActionPressed,
      duration: duration,
    );
  }

  /// Show warning snackbar
  static void showWarning({
    required BuildContext context,
    required String message,
    Duration duration = const Duration(seconds: 4),
  }) {
    show(
      context: context,
      message: message,
      type: SnackbarType.warning,
      duration: duration,
    );
  }

  /// Show info snackbar
  static void showInfo({
    required BuildContext context,
    required String message,
    Duration duration = const Duration(seconds: 3),
  }) {
    show(
      context: context,
      message: message,
      type: SnackbarType.info,
      duration: duration,
    );
  }

  /// Get colors for snackbar type
  static (IconData, Color, Color) _getColors(SnackbarType type) {
    switch (type) {
      case SnackbarType.success:
        return (Icons.check_circle, AppTheme.secondaryColor, Colors.white);
      case SnackbarType.error:
        return (Icons.error, AppTheme.errorColor, Colors.white);
      case SnackbarType.warning:
        return (
          Icons.warning_amber_rounded,
          AppTheme.warningColor,
          Colors.white,
        );
      case SnackbarType.info:
        return (Icons.info, AppTheme.infoColor, Colors.white);
    }
  }
}
